### DISCUSSION TITLE
what do you want to see in JSBattle?

### DISCUSSION THREAD

**Discussion by chinchang** [cssbattle/forum#53] [Ideas]
We have been brainstorming a lot internally about how the gameplay for JSBattle should look like. But we realized the best way to come up with a conclusion would be to ask you, the players, how they would want to play! 

Of course, the most trivial gameplay could be to recreate CSSBattle like targets in JavaScript, probably just using Canvas. But would that be fun for JS developers? Would that be helpful in learning JS? There are a lot of things we want to weigh the ideas on. And we are open to just about anything at all.

So, let us know your ideas on how JSBattle should be. We are listening.

**Comment by Vuurvos1** [comment]
While something like recreating the CSSBattle targets with JavaScritp might be fun it might feel more like learning Canvas than learning JavaScript.
What first came to mind for me is something like what they do for Competetive Programming were given a certain set of inputs you create an output with a set of constraints. You could then give points based on code length or speed.

**Comment by federicomichela** [comment]
I think the aim should be to learn js basic tricks like for CSSBattles. Things like alternative ways to "wait" (request animation frames, setTimeout, promises), alternative ways to loop (for, for-in, for-of, forEach, recursion), alternative ways to manipulate strings (String prototype methods, string interpolated, regex)...then maybe getting familiar on DOM manipulation through js like creating and handling some audio, videos, animations even (web animation API Vs applying styles / classes dynamically)...

About measuring I agree with the previous answer, definitely speed, possibly code length and measuring memory usage.
